deepchecks/core/check_result.py

```diff
@@ -146,40 +159,71 @@ def __init__(
             if not isinstance(item, (str, pd.DataFrame, Styler, Callable, BaseFigure)):
                 raise DeepchecksValueError(f'Can\'t display item of type: {type(item)}')
 
-    def to_widget(
-        self,
-        unique_id: Optional[str] = None,
-        show_additional_outputs: bool = True
-    ) -> Widget:
-        """Return CheckResult as a ipywidgets.Widget instance.
+    def _get_metadata(self, with_doc_link: bool = False):
+        check_name = self.check.name()
+        parameters = self.check.params(show_defaults=True)
+        header = self.get_header()
+        return {'name': check_name, 'params': parameters, 'header': header,
+                'summary': get_docs_summary(self.check, with_doc_link=with_doc_link)}
 
-        Parameters
-        ----------
-        unique_id : str
-            The unique id given by the suite that displays the check.
-        show_additional_outputs : bool
-            Boolean that controls if to show additional outputs.
+    def get_header(self) -> str:
```

Comment:
remove methods that are already in BaseCheckResult

Reply:
removed